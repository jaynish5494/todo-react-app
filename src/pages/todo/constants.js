export const MAP_TODO = 'todo/MAP_TODO';
export const ADD_TODO = 'todo/ADD_TODO';
export const GET_TODOS = 'todo/GET_TODOS';
export const DELETE_TODO = 'todo/DELETE_TODO';
export const EDIT_TODO = 'todo/EDIT_TODO';
