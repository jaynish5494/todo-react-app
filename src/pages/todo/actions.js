import { ADD_TODO, GET_TODOS, DELETE_TODO, EDIT_TODO } from './constants';

export const addTodo = task => ({
  type: ADD_TODO,
  task
});

export const getTodos = () => ({
  type: GET_TODOS
});

export const deleteTodo = id => ({
  type: DELETE_TODO,
  id
});

export const editTodo = (todo, id) => ({
  type: EDIT_TODO,
  todo,
  id
});
