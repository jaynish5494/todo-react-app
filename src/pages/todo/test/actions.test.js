import * as actions from '../actions';
import { ADD_TODO, GET_TODOS, DELETE_TODO, EDIT_TODO } from '../constants';

describe('todo actions', () => {
  it('should be defined', () => {
    expect(actions).toMatchSnapshot();
  });
  it('should create an action to add todo', () => {
    const task = 'Call obama';
    const expected = {
      type: ADD_TODO,
      task
    };
    expect(actions.addTodo(task)).toEqual(expected);
  });
  it('should create an action to get todo', () => {
    const expected = {
      type: GET_TODOS
    };
    expect(actions.getTodos()).toEqual(expected);
  });
  it('should create an action to delete todo', () => {
    const id = '123456789gfdxcvghj';
    const expected = {
      type: DELETE_TODO,
      id
    };
    expect(actions.deleteTodo(id)).toEqual(expected);
  });
  it('should create an action to edit todo', () => {
    const todo = 'Call trump';
    const id = '12345678dfghjkhg';
    const expected = {
      type: EDIT_TODO,
      todo,
      id
    };
    expect(actions.editTodo(todo, id)).toEqual(expected);
  });
});
