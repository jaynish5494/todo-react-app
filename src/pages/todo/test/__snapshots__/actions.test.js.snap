// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo actions should be defined 1`] = `
Object {
  "addTodo": [Function],
  "deleteTodo": [Function],
  "editTodo": [Function],
  "getTodos": [Function],
}
`;
